packages/secure_storage/amplify_secure_storage_dart/lib/src/utils/file_key_value_store.dart

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:file/file.dart' as pkg_file;
+import 'package:file/local.dart' as local_file;
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as path;
+
+/// {@template amplify_secure_storage_dart.file_key_value_store}
+/// A Key-Value Store that stores data in the provided file.
+/// {@endtemplate}
+@internal
+class FileKeyValueStore {
+  /// {@macro amplify_secure_storage_dart.file_key_value_store}
+  FileKeyValueStore({required this.directory, required this.fileName});
```

Comment:
I would do something like:

```dart
FileKeyValueStore({
  required this.directory, 
  required this.fileName,
  @visibleForTesting this.fs = const local_file.LocalFileSystem(),
});

@visibleForTesting
final pkg_file.FileSystem fs;
```

Reply:
Nice. Will update this and add some tests for this 👍